core/src/main/java/com/linecorp/armeria/server/UncaughtExceptionsServerErrorHandler.java

```diff
@@ -0,0 +1,113 @@
+ * {@link UncaughtExceptionsServerErrorHandler} is used to catch those uncaught exceptions.
+ */
+public class UncaughtExceptionsServerErrorHandler implements ServerErrorHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(UncaughtExceptionsServerErrorHandler.class);
+    private final ServerErrorHandler delegate;
+    private final LongAdder counter;
+    private final long intervalInSeconds;
+    private boolean isScheduled;
+    @Nullable
+    private Throwable lastThrownException;
+    @Nullable
+    private ScheduledFuture<?> reportUncaughtExceptionsSchedule;
+
+    UncaughtExceptionsServerErrorHandler(ServerErrorHandler serverErrorHandler, long intervalInSeconds) {
+        delegate = serverErrorHandler;
+        this.intervalInSeconds = intervalInSeconds;
+        counter = new LongAdder();
+    }
+
+    /**
+     * If it's required to log uncaught exceptions, increase the counter and store last thrown exception.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        if (isScheduled && ctx.shouldLogUncaughtExceptions()) {
+            counter.increment();
+            lastThrownException = cause;
+        }
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    /**
+     * Schedules uncaught exceptions logging.
+     * @param workerGroup eventLoopGroup on which logging will be scheduled.
+     */
+    public void scheduleLogging(EventLoopGroup workerGroup) {
+        requireNonNull(workerGroup, "workerGroup");
+        if (!isScheduled) {
+            reportUncaughtExceptionsSchedule = workerGroup.scheduleAtFixedRate(
+                    this::logUncaughtExceptions, intervalInSeconds, intervalInSeconds, TimeUnit.SECONDS);
+            isScheduled = true;
+        }
+    }
+
+    /**
+     * Unschedule uncaught exceptions logging.
+     */
+    public void unScheduleLogging() {
+        if (isScheduled && reportUncaughtExceptionsSchedule != null) {
+            reportUncaughtExceptionsSchedule.cancel(true);
+            isScheduled = false;
+        }
+    }
+
+    private void logUncaughtExceptions() {
+        logger.warn("Observed {} uncaught exceptions in last {} seconds. " +
+                    "If you don't see error logs please use LoggingService decorator.",
+                    counter.sumThenReset(), intervalInSeconds);
+        if (lastThrownException != null) {
+            logger.warn("Last thrown exception is: ", lastThrownException);
+        }
+        lastThrownException = null;
```

Comment:
- Should not log anything if count is 0
- Could log only in one log message, rather than two? But please handle the potential case where `lastThrownException == null && count != 0`.

Reply:
I think it's better to have one log message to handle edge cases. 👍